File: lib/platform_specifics/android_styles/big_text_style_information.dart
import 'package:flutter_local_notifications/platform_specifics/android_styles/style_information.dart';

class BigTextStyleInformation extends StyleInformation {
  final String bigText;
  final String contentTitle;
  final String summaryText;
  final bool htmlFormatBigText;
  final bool htmlFormatContentTitle;
  final bool htmlFormatSummaryText;

  BigTextStyleInformation(this.bigText,
      {this.htmlFormatBigText = false,
      this.contentTitle,
      this.htmlFormatContentTitle = false,
      this.summaryText,
      this.htmlFormatSummaryText = false})
      : super();

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'bigText': bigText,
      'htmlFormatBigText': htmlFormatBigText,
      'contentTitle': contentTitle,
      'htmlFormatContentTitle': htmlFormatContentTitle,
      'summaryText': summaryText,
      'htmlFormatSummaryText': htmlFormatSummaryText
    };
  }
}

File: android/src/main/java/com/dexterous/flutterlocalnotifications/BigTextStyleInformation.java
package com.dexterous.flutterlocalnotifications;

public class BigTextStyleInformation extends StyleInformation {
    public String bigText;
    public Boolean htmlFormatBigText;
    public String contentTitle;
    public Boolean htmlFormatContentTitle;
    public String summaryText;
    public Boolean htmlFormatSummaryText;

    public BigTextStyleInformation(String bigText, Boolean htmlFormatBigText, String contentTitle, Boolean htmlFormatContentTitle, String summaryText, Boolean htmlFormatSummaryText) {
        this.bigText = bigText;
        this.htmlFormatBigText = htmlFormatBigText;
        this.contentTitle = contentTitle;
        this.htmlFormatContentTitle = htmlFormatContentTitle;
        this.summaryText = summaryText;
        this.htmlFormatSummaryText = htmlFormatSummaryText;
    }
}
